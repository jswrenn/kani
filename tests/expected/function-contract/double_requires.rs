// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT

#[kani::ensures(result <= x || result == 1)]
#[kani::ensures(result != 0)]
#[kani::requires(x < u32::MAX)]
fn reduce1(x: u32, y: u32) -> u32 {
    x + 1
}

#[kani::ensures(result <= x || result == 1)]
#[kani::ensures(result != 0)]
#[kani::requires(x < u32::MAX)]
fn reduce2(x: u32, y: u32) -> u32 {
    x
}

#[kani::proof]
fn main2() {
    reduce2(kani::any(), kani::any());
}

#[kani::proof]
fn main() {
    reduce1(kani::any(), kani::any());
}
