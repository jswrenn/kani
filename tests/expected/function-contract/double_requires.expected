double_requires::reduce1.missing_definition \
	 - Status: FAILURE \
	 - Description: "Function `double_requires::reduce1` with missing definition is unreachable"

VERIFICATION:- FAILED

double_requires::reduce2.missing_definition \
	 - Status: FAILURE \
	 - Description: "Function `double_requires::reduce2` with missing definition is unreachable"
